tests/promise/promise.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals describe, it, expect, CKEDITOR, window */
+
+'use strict';
+
+describe( 'Promise', function() {
```

Comment:
> Well, the test is doing the most basic thing: it creates a Promise and consumes it. I've made it async just to make it more realistic and the setTimeout inside then() is mandatory.

We don't need more realism here - we're not testing our code - we're testing either native implementations (which we have to believe work correctly) or polyfills (which we have to believe work correctly too).

Besides that, I don't think we should use timeouts if we don't need to. Here you can do:

```
return new Promise( function( resolve ) {
    resolve( 1 );
} ).then( function( value ) {
    expect( value ).to.equal( 1 );
} );
```

Or even:

```
return Promise.resolve( 1 ).then( function( value ) {
    expect( value ).to.equal( 1 );
} );
```